### PR TITLE
Feature release/refactor impersonator

### PR DESCRIPTION
**Pull request by chtyim**
The first commit is just moving files from cdap-data-fabric to cdap-common + optimize import. 
Only the second and third commits are the actual changes.

### CODE REVIEW

File: cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright © 2014-2015 Cask Data, Inc.
+ * Copyright © 2014-2016 Cask Data, Inc.
```
Comment (anwar6953) [inline cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java:2]: oh your IDE automatically does this? I need to make mine do this automatically, too...

File: cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverTest.java [hunk 2]
```diff
@@ -108,7 +109,7 @@ public static void startLogSaver() throws Exception {
     cConf.set(LoggingConfiguration.LOG_BASE_DIR, logBaseDir);
 
     injector = KAFKA_TESTER.getInjector();
-    impersonator = injector.getInstance(Impersonator.class);
+    impersonator = injector.getInstance(DefaultImpersonator.class);
```
Comment (anwar6953) [inline cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverTest.java]: we're still not binding Impersonator to anything, and still getting the concrete class.
Wasn't this causing issues?

File: cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java (hunk 1 above)
Comment (chtyim) [inline cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java:2]: It's manual....... And I won't be your IDE

File: cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverTest.java (hunk 2 above)
Comment (chtyim) [inline cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverTest.java]: This is unit test, doesn't matter. However, I think it's better not to do this. Will change

File: cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java (hunk 1 above)
Comment (chtyim) [inline cdap-watchdog/src/test/java/co/cask/cdap/logging/write/LogCleanupTest.java:2]: Although it is not too difficult. Just do a find and replace over changed files (IntelliJ has this option)

File: cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverTest.java (hunk 2 above)
Comment (anwar6953) [inline cdap-watchdog/src/test/java/co/cask/cdap/logging/save/LogSaverTest.java]: fair enough

**Comment by anwar6953** [comment]
Changes lgtm

**Comment by chtyim** [comment]
Build passed http://builds.cask.co/browse/CDAP-RUT124

The second commit merely change the `Injector.getInstance` call in unit tests. It shouldn't affect any functionality. Affected tests are verified locally.

**Pull request merged**